agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2413,6 +2432,163 @@ bool backhaul_manager::handle_ap_capability_query(ieee1905_1::CmduMessageRx &cmd
+        return ret;
+    }
+    for (size_t bssid_idx = 0; bssid_idx < ap_metrics_query_tlv->bssid_list_length(); bssid_idx++) {
+        auto bssid_tupl = ap_metrics_query_tlv->bssid_list(bssid_idx);
+        if (!std::get<0>(bssid_tupl)) {
+            LOG(ERROR) << "Failed to get bssid " << bssid_idx << " from AP_METRICS_QUERY";
+            return ret;
+        }
+        bssid.push_back(std::get<1>(bssid_tupl));
+        LOG(DEBUG) << "Received AP_METRICS_QUERY_MESSAGE, mid=" << std::hex << int(mid)
+                   << "  bssid " << std::get<1>(bssid_tupl);
+    }
+
+    for (auto socket : slaves_sockets) {
+        for (auto mac : bssid) {
+            int i = 0;
+            if (mac == socket->vaps_list.vaps[i].mac) {
+                LOG(DEBUG) << "Forwarding AP_METRICS_QUERY_MESSAGE message to son_slave, bssid: "
+                           << std::hex << network_utils::mac_to_string(mac);
+
+                auto forward =
+                    cmdu_tx.create(mid, ieee1905_1::eMessageType::AP_METRICS_QUERY_MESSAGE);
+                if (!forward) {
+                    LOG(ERROR) << "Failed to create AP_METRICS_QUERY_MESSAGE";
+                }
+
+                auto query = cmdu_tx.addClass<wfa_map::tlvApMetricQuery>();
+                if (!query) {
+                    LOG(ERROR) << "Failed addClass<wfa_map::tlvApMetricQuery>";
+                }
+
+                if (!query->alloc_bssid_list(1)) {
+                    LOG(ERROR) << "Failed allocate memory for bssid_list";
+                }
+
+                auto list         = query->bssid_list(0);
+                std::get<0>(list) = true;
+                std::get<1>(list) = mac;
+
+                if (!message_com::send_cmdu(socket->slave, cmdu_tx)) {
+                    LOG(ERROR) << "Failed forwarding AP_METRICS_QUERY_MESSAGE message to son_slave";
+                    ret = false;
+                    continue;
+                } else {
+                    ret = true;
+                    // Fill a query vector
+                    ap_metric_query.push_back({socket->slave, mac});
+                }
+            }
+            i++;
+        }
+    }
+    return ret;
+}
+
+bool backhaul_manager::handle_ap_metrics_response(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                                  const std::string &src_mac)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received AP_METRICS_RESPONSE_MESSAGE, mid=" << std::hex << int(mid);
+
+    auto ap_metric_tlv = cmdu_rx.getClass<wfa_map::tlvApMetric>();
+    if (!ap_metric_tlv) {
+        LOG(ERROR) << "Failed cmdu_rx.getClass<wfa_map::tlvApMetric>(), mid=" << std::hex
+                   << int(mid);
+        return false;
+    }
+
+    auto bssid_tlv = ap_metric_tlv->bssid();
+    auto mac       = std::find_if(
+        ap_metric_query.begin(), ap_metric_query.end(),
+        [&](sApMetricsQuery const &ap_metric_query) { return ap_metric_query.bssid == bssid_tlv; });
```

Comment:
It is better to be specific, just for clarity -
```suggestion
        [&bssid_tlv](sApMetricsQuery const &ap_metric_query) { return ap_metric_query.bssid == bssid_tlv; });
```
(However, I don't have strong opinion about this issue)